main.js

```diff
@@ -5,46 +5,57 @@
 // Import Apify SDK. For more information, see https://sdk.apify.com/
 const Apify = require('apify');
 const Worker = require('./src/worker');
+const { log } = require('./src/utils');
+const { ERROR_TYPES } = require('./src/consts');
 
 Apify.main(async () => {
   // Initialize state values
   const input = await Apify.getInput();
   const { startupJobsToken, jazzHRToken } = input;
   // Open a named dataset
   const dataset = await Apify.openDataset('RECORDS');
-  const store = await Apify.openKeyValueStore('state');
-  const lastApplication = await store.getValue('LAST_APPLICATION') || {};
-  const unresolvedErrors = await store.getValue('UNRESOLVED_ERRORS') || [];
+  const store = await Apify.openKeyValueStore('prev_transfer_info');
+  let lastApplication = await store.getValue('LAST_APPLICATION') || {};
+  const resolvableErrors = await store.getValue('RESOLVABLE_ERRORS') || [];
 
   const worker = await Worker.create(startupJobsToken, jazzHRToken);
   const { items: stateRecords } = await dataset.getData();
 
   try {
     // Initialize values from state
+    log.info('Initiate state');
     const initialRecords = await worker.getNewRecords(stateRecords);
     await dataset.pushData(initialRecords);
     const { items: initializedRecords } = await dataset.getData();
 
     // Get new startupjobs application
+    log.info('Get startupjobs applications');
     const postable = await worker.getNewApplications(initializedRecords, lastApplication);
 
     // Post to jazzHR
-    const remainingPostErrors = await worker.resolvePostErrors(unresolvedErrors);
+    log.info('Transfering applications', { total: postable.length, applications: postable });
     const newPostErrors = await worker.postNewApplications(postable);
+    const allResolvableErrors = [...resolvableErrors, ...newPostErrors];
+    log.info('Retrying resolvable errors from this and previous run', { total: allResolvableErrors.length, errors: allResolvableErrors });
+    const remainingPostErrors = await worker.resolvePostErrors(allResolvableErrors);
 
     // Update state
+    log.info('Updating actor state for next runs');
     const newRecords = await worker.getNewRecords(initializedRecords);
     await dataset.pushData(newRecords);
-    await store.setValue('UNRESOLVED_ERRORS', [...remainingPostErrors, ...newPostErrors]);
-    if (postable.length > 0) await store.setValue('LAST_APPLICATION', postable[0]);
-
-    const stats = {
-      totalRecords: initializedRecords.length + newRecords.length,
-      postableApplications: postable.length,
-      unresolvedErrors: remainingPostErrors.length + newPostErrors.length,
-    };
-    console.log('Stats: ', stats);
+    await store.setValue('RESOLVABLE_ERRORS', remainingPostErrors);
+    if (postable.length > 0) [lastApplication] = postable;
```

Comment:
I use default airbnb linting rules which prohibits page++ .. I was too lazy to override the rule so I just left it like that .. As for concurrency I will create constants but there will have to be 2 separate ones for startupjobs and jazzHR since their APIs react differently